blockindex/contractstaking/cache.go

```diff
@@ -421,3 +449,15 @@ func (s *contractStakingCache) mergeDelta(delta *contractStakingDelta) error {
 	}
 	return nil
 }
+
+func (s *contractStakingCache) validateHeight(height uint64) error {
+	// means latest height
+	if height == 0 {
+		return nil
+	}
+	// indexer only store latest height
+	if height != s.height {
```

Comment:
should be `height >= s.height`?

in `action/protocol/poll/util.go`, it is called with epoch start height
```
func createPostSystemActions(ctx context.Context, sr protocol.StateReader, p Protocol) {
    xxx
    l, err := p.CalculateCandidatesByHeight(ctx, sr, epochHeight)
    xxx
}
```

Reply:
using sr.Height() instread of input height could avoid this